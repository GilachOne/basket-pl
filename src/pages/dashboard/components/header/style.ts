import styled from '@emotion/styled';

export const Header = styled.header`
    height: 64px;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: ${props => props.theme.colors.background.main};
`;

export const HeaderText = styled.p`
    font-family: 'Roboto';
    font-style: bold;
    font-size: 28px;
    color: ${props => props.theme.colors.text.main};
`;

export const Login = styled.p`
    font-family: 'Roboto';
    font-style: normal;
    font-size: 14px;
    color: ${props => props.theme.colors.accent.light};
`;

export const LogOut = styled.button`
    background: none;
    border: none;
    cursor: pointer;
    :focus {
        outline: none;
    }
`;

export const Wrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    right: 24px;
    position: absolute;
`;
