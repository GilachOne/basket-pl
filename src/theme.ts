export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
