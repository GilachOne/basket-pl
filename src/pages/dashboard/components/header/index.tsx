import React from 'react';
import { Header, HeaderText } from './style';
import { useTranslation } from 'react-i18next';

import { logout } from '../../../../public';
import { removeToken } from '../../../../utils/token';
import { URLs } from '../../../../__data__/urls';

import { Wrapper, LogOut } from './style';

export const HeaderComponent = () => {
    const { t } = useTranslation();

    const logOut = () => {
        removeToken();
        location.href = URLs.signIn.url;
    };

    return (
        <Header>
            <HeaderText>{t('basket.dashboard.header.title')}</HeaderText>
            <Wrapper>
                {/*TODO: вывести логин юзера <Login>{getToken()}</Login> */}
                <LogOut
                    onClick={logOut}
                    data-testid={t('basket.dashboard.header.logout')}
                >
                    <img
                        src={logout}
                        title={t('basket.dashboard.header.logout')}
                    />
                </LogOut>
            </Wrapper>
        </Header>
    );
};
