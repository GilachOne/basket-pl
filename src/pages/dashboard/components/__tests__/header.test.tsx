import React from 'react';
import { describe, it, expect } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { BrowserRouter } from 'react-router-dom';

import { HeaderComponent } from '../header';
import { theme } from '../../../../theme';

describe('Dashboard Header', () => {
    it('Header', () => {
        render(
            <ThemeProvider theme={theme}>
                <BrowserRouter>
                    <HeaderComponent />
                </BrowserRouter>
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTestId('Выход'));

        expect(document.body).toMatchSnapshot();
    });
});
