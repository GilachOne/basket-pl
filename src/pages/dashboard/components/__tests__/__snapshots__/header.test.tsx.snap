// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Header Header 1`] = `
<body>
  <div>
    <header
      class="css-22fkfx"
    >
      <p
        class="css-1adnaza"
      >
        вМагазин
      </p>
      <div
        class="css-1v5wvrd"
      >
        <button
          class="css-z414q7"
          data-testid="Выход"
        >
          <img
            src="log-out.png"
            title="Выход"
          />
        </button>
      </div>
    </header>
  </div>
</body>
`;
